pkg/webhook/controlplane/ensurer.go

```diff
@@ -88,22 +91,28 @@ func ensureVolumes(ps *corev1.PodSpec) {
 	ps.Volumes = extensionswebhook.EnsureVolumeWithName(ps.Volumes, authnWebhookCertVolume)
 }
 
-func ensureKubeAPIServerCommandLineArgs(c *corev1.Container) {
+func ensureKubeAPIServerCommandLineArgs(c *corev1.Container, controllerConfig config.ControllerConfiguration) {
 	c.Command = extensionswebhook.EnsureStringWithPrefix(c.Command, "--cloud-provider=", "external")
 
 	// activate AuthN Webhook with mounted Webhook-Config
 	c.Command = extensionswebhook.EnsureStringWithPrefix(c.Command, "--authentication-token-webhook-config-file=", "/etc/webhook/config/authn-webhook-config.json")
```

Comment:
Good catch. I also surrounded these parts with a switch.
For now I think it's ok not to create individual functions because this would also increase the maintenance overhead if we diverge too much from the implementation of the other controllers. It will make more sense to do this when we have more things to do in there.